main_27_7_2018.sh

```diff
@@ -34,26 +38,50 @@ sample_accessions=("SRR1258218" "SRR1153470")
 ref_genome="ens_build38_release92"
 ref_genome_url='ftp://ftp.ensembl.org/pub/release-92/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.*.fa.gz' ## primaly assembly has unplaced contigs while the vcf has only chromosomes
+ref_annot='ftp://ftp.ensembl.org/pub/release-92/gtf/homo_sapiens/Homo_sapiens.GRCh38.92.gtf.gz'
 
 ## download GIAB data using fastq-dump
 cd $work_dir/data
 module load GNU/4.4.5
 module load SRAToolkit/2.8.2
-for sample_accession in ${sample_accession[@]};do
-  fastq-dump --defline-seq '@$sn[_$rn]/$ri' --split-files $sample_accession
-done
+wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR115/SRR1153470/SRR1153470.sra
+wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR125/SRR1258218/SRR1258218.sra
 wget $ref_vcf_url -O $work_dir/ref_vcf/$sample.vcf.gz
+gunzip $ref_vcf/$sample.vcf.gz#gunzip
 wget $ref_vcf_url.tbi -O $work_dir/ref_vcf/$sample.vcf.gz.tbi
 wget $ref_genome_url -P $work_dir/genomeDir
 gunzip -c $work_dir/genomeDir/Homo_sapiens.GRCh38.dna.chromosome.*.fa.gz > $work_dir/genomeDir/$ref_genome.fa
 rm $work_dir/genomeDir/Homo_sapiens.GRCh38.dna.chromosome.*.fa.gz 
 sed -i 's/^>\(.*\) dna.*/>chr\1/' $work_dir/genomeDir/$ref_genome.fa
+wget $ref_annot -O $anno_dir
+mv work_dir/annotation/pub/release-92/gtf/homo_sapiens/*.gtf.gz $work_dir/annotation #the annotation file downloaded in this sequence of folders
+gunzip $work_dir/annotation/*.gtf.gz
+
+#fastq-dump for SRR1153470 and SRR1258218
+fastq-dump --defline-seq '@$sn[_$rn]/$ri' --split-files $work_dir/data/SRR1153470.sra
+fastq-dump --defline-seq '@$sn[_$rn]/$ri' --split-files $work_dir/data/SRR1258218.sra
+rm $work_dir/data/SRR1153470.sra
+rm $work_dir/data/SRR1258218.sra
 
 ## create a subset sample for testing
 sample_accession="SRR1258218"
 head -n4000000 $work_dir/data/${sample_accession}_1.fastq > $work_dir/data/${sample_accession}_subset_1.fastq
 head -n4000000 $work_dir/data/${sample_accession}_2.fastq > $work_dir/data/${sample_accession}_subset_2.fastq
 ########################################################################################################
+
+#genome indexing
+STAR --runMode genomeGenerate --genomeDir $outDir --genomeFastaFiles $genomeDir/$ref_genome.fa --runThreadN 7
+#sample mapping
+cd $work_dir
+mkdir -p STAR
+STAR --genomeDir $outDir/ --readFilesIn $data/${sample_accession}_subset_1.fastq,$data/${sample_accession}_subset_2.fastq --outFileNamePrefix STAR/${sample_accession}_supset --sjdbGTFfile $anno_dir/* --runThreadN 7
+
+#MarkDuplicates (Picard)
+cd $work_dir
+mkdir -p picard
+#there is no output from this file
+java -jar $picard MarkDuplicates I=STAR/${sample_accession}_supsetAligned.out.bam O=picard/marked_duplicates_${sample_accession}_subset.bam M=picard/marked_dup_metrics_${sample_accession}_subset.txt
+
```

Comment:
As we discussed before, this block of code should be after line 63 and use only the variables defined in lines 3-10 and 59-61